### ISSUE TITLE
Suffix matching does not support Chinese characters

### ISSUE THREAD

**Issue by vampireslove** [aws/event-ruler#73]
## To Reproduce
     @Test
    public void testSuffixChineseMatch() throws Exception {
        Machine m = new Machine();
        String rule = "{\n" +
                "   \"status\": {\n" +
                "       \"weatherText\": [{\"suffix\": \"雨\"}]\n" +
                "    }\n" +
                "}";
        String eventStr ="{\n" +
                "  \"status\": {\n" +
                "    \"weatherText\": \"大雨\",\n" +
                "    \"pm25\": 23\n" +
                "  }\n" +
                "}";
        m.addRule("r1", rule);
        List<String> matchRules = m.rulesForJSONEvent(eventStr);
        assertEquals(1, matchRules.size());
    }

**Comment by baldawar** [comment]
Can confirm I see this. I suspect this is because of how we parse input

* https://github.com/aws/event-ruler/blob/main/src/main/software/amazon/event/ruler/ByteMachine.java#L698
* https://github.com/aws/event-ruler/blame/main/src/main/software/amazon/event/ruler/input/DefaultParser.java#L60

**Comment by baldawar** [comment]
Can verify wildcard works (see test below).  

```
@Test
    public void testSuffixChineseMatch() throws Exception {
        Machine m = new Machine();
        String rule = "{\n" +
                "   \"status\": {\n" +
                "       \"weatherText\": [{\"wildcard\": \"*雨\"}]\n" +
                "    }\n" +
                "}";
        String eventStr ="{\n" +
                "  \"status\": {\n" +
                "    \"weatherText\": \"大雨\",\n" +
                "    \"pm25\": 23\n" +
                "  }\n" +
                "}";
        m.addRule("r1", rule);
        List<String> matchRules = m.rulesForJSONEvent(eventStr);
        assertEquals(1, matchRules.size());
    }
```

**Comment by vampireslove** [comment]
"wildcard","prefix", I have tried, match Chinese is OK. But "exactly" and "suffix" won't work.

**Comment by vampireslove** [comment]
> Can verify wildcard works. This test passes.
> 
> ```
> @Test
>     public void testSuffixChineseMatch() throws Exception {
>         Machine m = new Machine();
>         String rule = "{\n" +
>                 "   \"status\": {\n" +
>                 "       \"weatherText\": [{\"wildcard\": \"*雨\"}]\n" +
>                 "    }\n" +
>                 "}";
>         String eventStr ="{\n" +
>                 "  \"status\": {\n" +
>                 "    \"weatherText\": \"大雨\",\n" +
>                 "    \"pm25\": 23\n" +
>                 "  }\n" +
>                 "}";
>         m.addRule("r1", rule);
>         List<String> matchRules = m.rulesForJSONEvent(eventStr);
>         assertEquals(1, matchRules.size());
>     }
> ```

"wildcard","prefix", I have tried, match Chinese is OK. But "exactly" and "suffix" won't work.

**Comment by vampireslove** [comment]
> 

"wildcard","prefix", I have tried, match Chinese is OK. But "exactly" and "suffix" won't work.

**Comment by vampireslove** [comment]
> Can verify wildcard works. This test passes.
> 
> ```
> @Test
>     public void testSuffixChineseMatch() throws Exception {
>         Machine m = new Machine();
>         String rule = "{\n" +
>                 "   \"status\": {\n" +
>                 "       \"weatherText\": [{\"wildcard\": \"*雨\"}]\n" +
>                 "    }\n" +
>                 "}";
>         String eventStr ="{\n" +
>                 "  \"status\": {\n" +
>                 "    \"weatherText\": \"大雨\",\n" +
>                 "    \"pm25\": 23\n" +
>                 "  }\n" +
>                 "}";
>         m.addRule("r1", rule);
>         List<String> matchRules = m.rulesForJSONEvent(eventStr);
>         assertEquals(1, matchRules.size());
>     }
> ```

"wildcard","prefix", I have tried, match Chinese is OK. But "exactly" and "suffix" won't work.

**Comment by baldawar** [comment]
That's good to know. Currently a bunch of us are focused on Ruler memory profile and improvements. Will look at this next. If there's a test case for exact match that you know for sure is failing let me know. We'll make sure to add that in when we push fixes (as long as you are okay with it).

**Comment by baldawar** [comment]
This has turned out to be tricker than I have bandwidth for. I'm going to switch to offering a smaller fix before switching back to `wildcard` support.

**Comment by baldawar** [comment]
Closing as fixed in v 1.2.1 https://github.com/aws/event-ruler/releases/tag/v1.2.1 

@vampireslove  thanks to spotting this.
